web/src/muted_topics_ui.js

```diff
@@ -22,7 +22,9 @@ export function rerender_for_muted_topic(old_muted_topics) {
     // We only update those topics which could have been affected, because
     // we want to avoid doing a complete rerender of the recent topics view,
     // because that can be expensive.
-    const current_muted_topics = user_topics.get_muted_topics();
+    const current_muted_topics = user_topics.get_user_topics_for_visibility_policy(
+        user_topics.all_visibility_policies.MUTED,
```

Comment:
can you please also update the `rerender_for_muted_topic` function to also work for unmute_topics as well? As I am working for #25124 in which clicking unmute icon in recent conversations doesn't update the topics unmute visbility_policy until the page reloads.
Maybe this will also help to render unmute topics messages on unmuting a topic in the All messages section without reloading which is being worked upon by @palashb01 in #25120.

Reply:
Hm, I think that should be done in a separate PR (or as a prep commit in the PR you're working on). -- It would make more sense in that PR to say 'why we made that change'.

The changes I have made here are related to making `get_muted_topics` support other visibility policies. -- It isn't related to `rerender_for_muted_topic`.

Reply:
sure, I will create a Prep PR soon.

Reply:
I have created this PR which refactors the `rerender_for_muted_topic` function to render unmuted topics without reloading. @Ddharmani3 , we don't need a separate PR for this. You can simply reference my PR.: https://github.com/zulip/zulip/pull/25120